src/sqlfluff/rules/L007.py

```diff
@@ -60,49 +67,148 @@ def _eval(self, context: RuleContext) -> LintResult:
             return LintResult(memory=memory)
+
+        expr = cast(Tuple[BaseSegment], parent_stack[-1].segments)
+        fixes = _generate_fixes(expr, anchor, operator_new_lines)
+        return LintResult(
+            anchor=anchor,
+            memory=memory,
+            description=description,
+            fixes=fixes,
+        )
+
+
+def _generate_fixes(
+    expr: Tuple[BaseSegment],
+    anchor: BaseSegment,
+    operator_new_lines: str,
+) -> List[LintFix]:
+    """Generate Fixes.
+
+    Take operator_new_lines==after as the default case.
+    Consider "before" as a modification on this behaviour.
+
+    """
+    fixes = [LintFix.delete(anchor)]
+    res = list(_get_surrounding_segments(expr, anchor))
+    if operator_new_lines == "before":
+        res = [list(reversed(els)) for els in reversed(res)]
+
+    change_list, anchor_list = res
+    insert_anchor = anchor_list[-1]
+    change_list = change_list[1:]
+    for el in change_list:
+        fixes.append(LintFix.delete(el))
+
+    change_list.append(anchor)
+    if operator_new_lines == "before":
+        # We do yet another reverse here,
+        # This could be avoided but makes all "changes" relate to "before" config state
+        change_list = list(reversed(change_list))
+
+    change_list = list(map(_copy_el, reversed(change_list)))
+    edit_type = "create_after" if operator_new_lines == "before" else "create_before"
+    fixes.append(
+        LintFix(
+            edit_type=edit_type,
+            edit=change_list,
+            anchor=insert_anchor,
+        )
+    )
+    return fixes
+
+
+def _copy_el(segment: BaseSegment):
+    """Safely duplicate segment without position markers."""
+    el = copy.deepcopy(segment)
+    el.pos_marker = None  # type: ignore
+    return el
+
+
+def _get_surrounding_segments(
+    expr: Tuple[BaseSegment],
+    symbol: BaseSegment,
+) -> Tuple[List[BaseSegment], List[BaseSegment]]:
+    """Create two lists of symbols surrounding operator.
+
+    1 from last code to operator/symbol,
+    2 from operator to next code
+
+    """
+    before_list: List[BaseSegment] = []
+    after_list: List[BaseSegment] = []
+    start_code: Optional[BaseSegment] = None
+    symbol_hit = False
+    for el in expr:
+        if el == symbol:
+            symbol_hit = True
+            continue
+        if el.is_code:
+            if symbol_hit:
+                # Exit condition, we have found code on the other side of the operator
+                after_list.append(el)
+                return before_list, after_list
+
+            start_code = el
+            before_list = []
+            after_list = []
+
+        if symbol_hit:
+            after_list.append(el)
+            continue
+
+        if start_code:
+            before_list.append(el)
+    # This code is unreachable if a valid expr and symbol/operator segment are passed.
+    raise Exception("Ivalid Expr: must contain binary_operator or comparison_operator")  # pragma: no cover
```

Comment:
Explicit no cover and Exception for this deliberately unreachable code seems reasonable right?